plugins/woocommerce/src/Internal/DataStores/Orders/DataSynchronizer.php

```diff
@@ -171,14 +188,179 @@ public function delete_database_tables() {
 	}
 
 	/**
-	 * Is the data sync between old and new tables currently enabled?
+	 * Is the real-time data sync between old and new tables currently enabled?
 	 *
 	 * @return bool
 	 */
 	public function data_sync_is_enabled(): bool {
 		return 'yes' === get_option( self::ORDERS_DATA_SYNC_ENABLED_OPTION );
 	}
 
+	/**
+	 * Is the background data sync between old and new tables currently enabled?
+	 *
+	 * @return bool
+	 */
+	public function background_sync_is_enabled(): bool {
+		$enabled = $this->data_sync_is_enabled();
+
+		/**
+		 * Toggle for background sync.
+		 *
+		 * This allows for background sync to be enabled even when real-time sync is disabled.
+		 *
+		 * @since 8.2.0
+		 *
+		 * @param bool $enabled True to enabled background sync.
+		 */
+		return apply_filters( 'woocommerce_custom_orders_table_background_sync_enabled', $enabled );
+	}
+
+	/**
+	 * Schedule an event to run background sync when the mode is not set to continuous.
+	 *
+	 * @return void
+	 */
+	private function schedule_background_sync() {
+		if ( ! $this->background_sync_is_scheduled() ) {
+			$interval = $this->get_background_sync_interval();
+
+			as_schedule_single_action(
+				time() + $interval,
+				self::BACKGROUND_SYNC_EVENT_HOOK,
+				array(),
+				'',
+				true
+			);
+
+			set_transient(
```

Comment:
why not use options instead of transient? transients can get ejected or expired. It's more of an option rather then a cache.

Reply:
🤔 What makes you think of this as more of an option? The mechanism seems pretty explicitly like a cache to me--we're saving a queried value to the database to avoid having to run the query every time. We _want_ the cached value to expire in case there's a pathway where the value doesn't get reset programmatically. Plus, the transient system uses a persistent object cache if it's available.

I suppose that we don't necessarily need to be setting the transient at this spot since it gets done in `background_sync_is_scheduled`...

Reply:
Made the toggle for background sync an option instead of a filter in 52dbc901af6fd6824b2358dcdf0599df1b697a27, which allows us to avoid the need for caching, because now the event only gets scheduled/unscheduled when the option value changes.